inst/include/dust/filter.hpp

```diff
@@ -27,7 +44,11 @@ filter(T * obj,
 
   if (save_trajectories) {
     state.trajectories.resize(obj->n_state(), n_particles, n_data);
-    obj->state(state.trajectories.value_iterator());
+
+    if (obj->step() <= obj->data().begin()->first) {
```

Comment:
Comment to explain what this is checking? The second term is a bit complex

Reply:
both a comment and a bit of self-documenting code to tidy it up. it's not bad, but not obvious when within the condition